server.ts

```diff
@@ -19,8 +19,149 @@ const {
 
 const agent = await login();
 
-function getEmbedImages(post: AtoprotoAPI.AppBskyFeedDefs.PostView) {
-  return AppBskyEmbedImages.isView(post.embed) ? post.embed.images : [];
+function getPost(
+  post: AtoprotoAPI.AppBskyFeedDefs.PostView,
+  fullMedia = false,
+) {
+  const author = post.author;
+
+  const text = (AppBskyFeedPost.isRecord(post.record))
+    ? processText(post.record)
+    : "";
+
+  const isReply = (AppBskyFeedPost.isRecord(post.record) && post.record.reply)
+    ? true
+    : false;
+
+  const media = (
+      //if post with media
+      AppBskyEmbedImages.isView(post.embed)
+    )
+    ? post.embed.images
+    : (
+        //if post with media and Quoted post with media
+        AppBskyEmbedRecordWithMedia.isView(post.embed) &&
+        AppBskyEmbedImages.isView(post.embed.media)
+      )
+    ? post.embed.media.images
+    : [];
+
+  const mediastr = (media.length > 0)
+    ? tag(
+      "div",
+      ...media.map((image) => {
+        return tag(
+          "figure",
+          `<img src="${
+            fullMedia ? image.fullsize : image.thumb
+          }"/><figcaption>${image.alt}</figcaption>`,
+          "<br>",
+        );
+      }),
+    )
+    : "";
+  return {
+    author: author,
+    uri: post.uri,
+    text: text,
+    isReply: isReply,
+    mediaarr: media,
+    media: mediastr,
+    quote: getQuotePost(post, fullMedia),
+  };
+}
+
+function getQuotePost(
+  post: AtoprotoAPI.AppBskyFeedDefs.PostView,
+  fullMedia = false,
+) {
+  const quotePost = (
+      //Text-only post with Quoted post with media
+      AppBskyEmbedRecord.isView(post.embed) &&
+      AppBskyEmbedRecord.isViewRecord(post.embed.record)
+    )
+    ? post.embed.record
+    : (
+        //Media post with quoted post with media
+        AppBskyEmbedRecordWithMedia.isView(post.embed) &&
+        AppBskyEmbedRecord.isViewRecord(post.embed.record.record)
+      )
+    ? post.embed.record.record
+    : undefined;
+
+  if (quotePost) {
```

Comment:
Please create a guard clause, instead of a long `if` and a short `else`.
```suggestion
  if (!quotePost) return undefined;
  // The process continues...
```